wire-schema/src/main/java/com/squareup/wire/schema/ProtoFile.java

```diff
@@ -174,6 +174,50 @@ ProtoFile retainAll(Schema schema, MarkSet markSet) {
     return result;
   }
 
+  public ProtoFile retainImports(List<ProtoFile> retained) {
+    ImmutableList.Builder<String> retainedImportsBuilder = ImmutableList.builder();
+    for (String path : imports) {
+      ProtoFile importedProtoFile = findProtoFile(retained, path);
+      if (importedProtoFile == null) continue;
+
+      if (importedProtoFile.types().isEmpty()
+          && importedProtoFile.services().isEmpty()
+          && importedProtoFile.extendList().isEmpty()) {
+
+        // If we extend a google protobuf type, we should keep the import.
+        if (path.startsWith("google/protobuf/")) {
```

Comment:
I think you can actually use an exact match: `path.equals("google/protobuf/descriptor.proto")`